lib/core/src/Cardano/Pool/DB/Sqlite.hs

```diff
@@ -214,6 +216,15 @@ newDBLayer trace fp = do
             fmap (poolRegistrationPoolId . entityVal) <$> selectList [ ]
                 [ Desc PoolRegistrationSlot ]
 
+        , putPoolMetadataRef = \poolId ref -> do
+            putMany [PoolMetadataQueue poolId (metadataURL ref) (metadataHash ref)]
+
+        , deletePoolMetadataRef = \poolId -> do
+            deleteWhere [PoolMetadataQueuePoolId ==. poolId]
```

Comment:
`deletePoolMetadataRef` seems unused. Can it be removed?

You're relying on overwrites to handle rollback? How about either
1. Rolling back `pool_metadata` and `pool_metadata_queue` with `SlotId` like normally?
2. Not including `PoolId` in  `pool_metadata` and `pool_metadata_queue`, but rather only using the metadata hash for the relationships, and primary key? (No fear for race conditions this way, I think)

Reply:
`deletePoolMetadataRef` is used by the worker fetching the metadata, coming in another PR.

> You're relying on overwrites to handle rollback?

Indeed. 

1. For the `pool_metadata_queue` it could make sense indeed, but not for the pool_metadata. Metadata aren't on-chain (although I believe your idea would be to assign them the slot id of the pool registration?). 

2. The `PoolId` is needed if you want to be able to map a particular pool to its metadata. Yet you're right, the metadata-hash feels like a better primary key. 

If we indeed go for the metadata-hash, it also reinforces my belief that rolling back pool metadata is awkward; we'd have to re-fetch them again even if the hash hasn't changed, which sounds like a waste of time / resources.

Reply:
Also, _why_ isn't `pool_metadata_queue` storing slot id and going through the normal rollback mechanism is because this table is ephemeral. Data in there are stored, and then removed. It's really just a queue (hence the name) and not used for long-term storage.

Reply:
👍 So let the metadata only be keyed by hash, and not pool id?

Reply:
Also: What are the reasons for keeping `PoolMetadataQueue` and `PoolRegistration` separate tables?

The worker _could_ select registration certificates whose metadata hashes are not in the metadata table?